modules/disko/disko-ab-partitions.nix

```diff
@@ -0,0 +1,181 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# !!! To utilize this partition scheme, the disk size must be >252G !!!
+#
+# This partition scheme contains three common partitions and two ZFS pools.
+# First LVM pool occupies 250G and second one occupies the rest of the disk space.
+# Some paritions are duplicated for the future AB SWupdate implementation.
+#
+# First three partitions are related to the boot process:
+# - boot : Bootloader partition
+# - ESP-A : (500M) Kernel and initrd
+# - ESP-B : (500M)
+#
+# First ZFS pool contains next partitions:
+# - root-A : (30G) Root FS
+# - root-B : (30G)
+# - vm-storage-A : (30G) Possible standalone pre-built VM images are stored here
+# - vm-storage-B : (30G)
+# - reserved-A : (10G) Reserved partition, no use
+# - reserved-B : (10G)
+# - gp-storage : (50G) General purpose storage for some common insecure cases
+# - recovery : (rest of the ZFS pool) Recovery factory image is stored here
+#
+# Second ZFS pool is dedicated for Storage VM completely.
+{pkgs, ...}: {
+  networking.hostId = "8425e349";
+  # A lot of memory required for some unknown reason
+  disko = {
+    memSize = 4096;
+    extraPostVM = ''
+      ${pkgs.zstd}/bin/zstd --compress $out/*raw
+      rm $out/*raw
+    '';
+    extraRootModules = ["zfs"];
+    devices = {
+      disk.disk1 = {
+        type = "disk";
+        imageSize = "15G";
+        content = {
+          type = "gpt";
+          partitions = {
+            boot = {
+              name = "boot";
+              size = "1M";
+              type = "EF02";
```

Comment:
```suggestion
              type = "EF02";
              priority = 1; # Needs to be first partition
```

Reply:
I got reports this is important for some EFI/BIOS implementations.

Reply:
Applied, thanks!